### PR TITLE
Assembly: Fix migrationScript2 error with bad joints

### PR DESCRIPTION
**Pull request by PaddleStroke**
Fix https://github.com/FreeCAD/FreeCAD/issues/16722

If you try to migrate an old file with a joint that does not have reference (Object1 or Object2 = None) then you get some error messages. This PR patches this issue

### CODE REVIEW

File: src/Mod/Assembly/JointObject.py [hunk 1]
```diff
@@ -454,51 +454,53 @@ def migrationScript(self, joint):
...
-
             joint.addProperty(
                 "App::PropertyXLinkSubHidden",
                 "Reference1",
                 "Joint Connector 1",
                 QT_TRANSLATE_NOOP("App::Property", "The first reference of the joint"),
             )
+            
+            if joint.Object1 is not None:
+                obj = joint.Object1[0]
```
Comment (luzpaz) [inline src/Mod/Assembly/JointObject.py:465]: trailing whitespace

Comment (PaddleStroke) [reply src/Mod/Assembly/JointObject.py:465]: Well that's weird I tried to remove it but the commit seems empty.

Comment (PaddleStroke) [reply src/Mod/Assembly/JointObject.py]: It's done.
Trailing whitespaces warnings is a waste of time IMO...

Comment (chennes) [reply src/Mod/Assembly/JointObject.py]: *All* formatting discussions are a waste of time - that's why we have pre-commit.

**Comment by chennes** [comment]
Backported to 1.0

**Pull request merged**